gmp/PKGBUILD

```diff
@@ -11,21 +11,45 @@ url="https://gmplib.org/"
 license=('LGPL3')
 install=gmp.install
 source=(https://gmplib.org/download/gmp-${pkgver}/gmp-${pkgver}.tar.xz{,.sig}
-        gmp-5.1.2-msys2.patch)
-sha256sums=('87b565e89a9a684fe4ebeeddb8399dce2599f9c9049854ca8c0dfbdea0e21912'
+        gmp-6.2.0-1.src.patch
+        gmp-6.2.0-msys2.patch)
+sha256sums=('258e6cd51b3fbdfc185c716d55f82c08aff57df0c6fbd143cf6ed561267a1526'
             'SKIP'
-            'c1ff80a8643a362f44be680a5430dca6f4ec16d84a4068d344a24b8d181e5436')
-validpgpkeys=('5972C48224C1E7E62AA0275D1150EE33C1A000B0')   # Marco Bodrato
+            'f8a7f986f8d646ef6ce2e3b8312ad3a6cc9e219e59d712b3ff30de995c857be2'
+            'fd1330261fb85c0b2b820535b58084a2ce6cfafe9a953cb9cfaa4dd46b4ab6c3')
+validpgpkeys=('5972C48224C1E7E62AA0275D1150EE33C1A000B0'   # Marco Bodrato
+              '343C2FF0FBEE5EC2EDBEF399F3599FF828C67298')   # Niels Möller
+
+# Helper macros to help make tasks easier #
+apply_patch_with_msg() {
+  for _patch in "$@"
+  do
+    msg2 "Applying $_patch"
+    patch -Nbp1 -i "${srcdir}/$_patch"
+  done
+}
+
+apply_patch_p2_with_msg() {
+  for _patch in "$@"
+  do
+    msg2 "Applying $_patch"
+    patch -Nbp2 -i "${srcdir}/$_patch"
+  done
+}
 
 prepare() {
   cd "${srcdir}/${pkgname}-${pkgver}"
-  patch -p1 -i ${srcdir}/gmp-5.1.2-msys2.patch
+  apply_patch_p2_with_msg gmp-6.2.0-1.src.patch
+  apply_patch_with_msg gmp-6.2.0-msys2.patch
 
   autoreconf -fi
 }
 
 build() {
   cd "${srcdir}/${pkgname}-${pkgver}"
+# get rid of those pesky declspec warnings
+  CFLAGS+=" -Wno-attributes" \
```

Comment:
Doesn't this hide potential bugs? What's the point?

Reply:
That's something Cygwin was doing.  We are probably not talking about errors but warnings.